packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2704,14 +2710,123 @@ class _Location {
   }
 }
 
+bool _isDebugCreator(DiagnosticsNode node) => node is DiagnosticsDebugCreator;
+
+/// Transformer to parse and gather information about [DiagnosticsDebugCreator].
+///
+/// This function will be registered to [FlutterErrorDetails.propertiesTransformers]
+/// in [WidgetsBinding.initInstances].
+List<DiagnosticsNode> transformDebugCreator(List<DiagnosticsNode> properties) {
+  if (properties.every((DiagnosticsNode node) => !_isDebugCreator(node)))
```

Comment:
if this was an iterator -> iterator method, you could just replace this whole method with a `.map` or `.expand` or similar (or a for loop with `yield`/`yield*`s).

Reply:
If we delegate one layer of iterator every time we called transformDebugCreator, do we need to worry about if we have a lot of transformDebugCreator that we end up with 10+ of delegations?

Reply:
We talked about this offline. I tried to convert this code to what it would look like using the iterator approach. I haven't tested it, so it could be very wrong, but here's what I came up with:
```dart
Iterable<DiagnosticsNode> transformDebugCreator(Iterable<DiagnosticsNode> properties) sync {
  final List<DiagnosticsNode> pending = <DiagnosticsNode>[];
  bool foundStackTrace = false;
  for (DiagnosticsNode node in properties) {
    if (!foundStackTrace && node is DiagnosticsStackTrace)
      foundStackTrace = true;
    if (_isDebugCreator(node))
      final DebugCreator debugCreator = node.value;
      final Element element = debugCreator.element;
      yield* _describeRelevantUserCode(element);
    } else {
      if (foundStackTrace) {
	pending.add(node);
      } else {
	yield node;
      }
    }  
  }  
  yield* pending;
}
```